npeps/npep-182-cidr-object-peer.md

```diff
@@ -0,0 +1,195 @@
+This also ensures reference of same CIDR group peer from ANP and BANP
+stays consistent and any changes to the list of CIDR blocks only involves
+editing the object itself and not the rules in the policy.
+
+### User Stories for CIDRGrouping
+
+* As a cluster admin I want to be able to create admin network policies that
+match a dynamic set of external IPs (e.g. set of VMs or set of directly reachable
+Pods in another cluster). I may not be able to use FQDN rules for that due to
+TTL being too long or simply lack of DNS service discovery in an external system.
+As a cluster admin, I would create CIDR group resource and a BGP controller that
+would manage it. The mapping between BGP communities and CIDR group resource names
+is a BGP controller configuration (e.g. annotation on the CIDR group resource).
+The speed of IP churn is bounded by the BGP advertisement interval and can be
+further reduced by the BGP controller implementation.
+
+* As a cluster administrator I want to to ensure that pods can reach
+commonly-used databases under my control but outside Kubernetes. Many but
+not all applications in my environment rely on these databases. I want to
+delegate writing network policy for this traffic to namespace owners.
+
+Example: As a cluster administrator I define a CIDR group that defines
+a set of RDS instances that is used across multiple apps. The owners of
+namespaceA and namespaceB can then define policies that allow traffic to
+this group of RDS instances, and they reference the instances by CIDR group.
+As a cluster administrator I can migrate the database infrastructure and
+update the CIDR group independently of the namespace owners. The applications
+in namespaceC do not use this infrastructure, so the cluster administrator
+and the owners of namespaceC do not need to think about network policy
+for apps in namespaceC.
+
+NOTE: Second use case is not possible today using NetworkPolicy resource
+since we only have `ipBlocks` as a peer however this is definitely a useful
+case to keep in mind for having a CIDR Group.
+
+## API
+
+This NPEP Proposes to add a new `CIDRGroup` object:
+
+```
+// CIDRGroup defines a group of CIDR blocks that can be referred to from
+// AdminNetworkPolicy and BaselineAdminNetworkPolicy resources.
+// +kubebuilder:validation:MaxProperties=1
+// +kubebuilder:validation:MinProperties=1
+type CIDRGroup struct {
+	// cidrs is the list of network cidrs that can be used to define destinations.
+	// A total of 25 CIDRs will be allowed in each CIDRGroup instance.
+	// ANP & BANP APIs may use the .spec.egress.to.networks.cidrGroups selector
+	// to select a set of cidrGroups.
+	//
+	// +optional
+	// +kubebuilder:validation:MinItems=1
+	// +kubebuilder:validation:MaxItems=25
+	cidrs []CIDR `json:"cidrs,omitempty"
+}
+```
+
+In order to ensure it is coexisting with inline CIDR peers without confusion,
+we propose to change the type of `networks` peer from `string` to a struct of type
+`NetworkPeer`:
```

Comment:
Given the API is headed for beta, would this breaking change be OK?  I'd have thought we want to add a new top-level peer type to make it easier to adopt in non-breaking fashion.